src/main/java/org/sonarlint/intellij/ui/ruledescription/RuleLanguages.kt

```diff
@@ -24,46 +24,42 @@ import com.intellij.openapi.fileTypes.FileTypeRegistry
 import com.intellij.openapi.fileTypes.UnknownFileType
 import org.sonarsource.sonarlint.core.commons.Language
 
-enum class RuleLanguages(val language: Language, val fileType: String?) {
+// ObjectiveC file type is available in CLion, and C++ is available in IDEA
+private val cFamilyLanguagesFileTypesByPriority = arrayOf("ObjectiveC", "C++")
 
-    ABAP(Language.ABAP, null),
-    APEX(Language.APEX, null),
-    C(Language.C, "C++"),
+// TypeScript file type is available in WebStorm, and JavaScript is available in IDEA
+private val typescriptFileTypesByPriority = arrayOf("TypeScript", "JavaScript")
+
+enum class RuleLanguages(private val language: Language, private vararg val fileTypesByPriorityOrder: String) {
+
+    C(Language.C, *cFamilyLanguagesFileTypesByPriority),
     CLOUD_FORMATION(Language.CLOUDFORMATION, "YAML"),
-    COBOL(Language.COBOL, null),
-    CPP(Language.CPP, "C++"),
+    CPP(Language.CPP, *cFamilyLanguagesFileTypesByPriority),
     CSH(Language.CS, "C#"),
     CSS(Language.CSS, "CSS"),
-    DOCKER(Language.DOCKER, null),
+    DOCKER(Language.DOCKER, "Dockerfile"),
```

Comment:
In which IDE is `Dockerfile` file type supported?

Reply:
Webstorm. I plan to verify the syntax highligthing next week during IaC integration